helix-lock/src/main/java/org/apache/helix/lock/helix/ZKDistributedNonblockingLock.java

```diff
@@ -144,19 +314,91 @@ public ZNRecord update(ZNRecord current) {
       // If no one owns the lock, allow the update
       // If the user is the current lock owner, allow the update
       LockInfo curLockInfo = new LockInfo(current);
-      if (!(System.currentTimeMillis() < curLockInfo.getTimeout()) || isCurrentOwner()) {
+      if (System.currentTimeMillis() > curLockInfo.getTimeout() || isCurrentOwner(curLockInfo)) {
```

Comment:
What if there is no owner at first beginning? Even no ZNode exist?

Reply:
Yeah, there won't be Znode, and the input value is null.

Reply:
Sorry. I am asking what if there is no ZNode, will this function breaking? Shall we create one? Because creating could also have race condition happening. How we handle the create fail.

Reply:
Yeah, it should work. This is the design in MVP, and I think it has been tested by customers. 
The logic is, if there is no znode existing, we create LockInfo with default value, and take that as current value, update that value with user's requested lock information.

Reply:
Also regarding the above question `Are we notifying them or we give them callback and process clean up here?`
Seems I cannot directly reply, so just answer here. We provide an interface of cleanup, and clients should implement that cleanup callback.